lib/elixir/lib/kernel/utils.ex

```diff
@@ -94,9 +92,14 @@ defmodule Kernel.Utils do
      Module.get_attribute(module, :derive)}
   end
 
+  def defstruct(module, fields) when is_atom(module) do
+    raise ArgumentError, "struct fields definition must be list, got: #{inspect fields}"
+  end
+
   @doc """
   Announcing callback for defstruct.
   """
+  @spec announce_struct(mod) :: (:ok | {:struct_available, mod}) when mod: module
```

Comment:
No need to spec this function. This is a callback for internal implementation details.